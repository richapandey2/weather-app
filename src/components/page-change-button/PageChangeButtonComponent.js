import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Grid, Fab } from "@material-ui/core";
import {
  PAGE_BUTTON_ICON_PREVIOUS,
  PAGE_BUTTON_ICON_NEXT,
} from "../../constants/Constants";

function PageChangeButtonComponent(props) {
  const { pageDetails, pageIndex, handlePageButton } = props;
  return (
    <Grid
      data-test-id="container"
      container
      spacing={4}
      style={{ justifyContent: "space-between" }}
    >
      <Grid item>
        {pageIndex > 0 && (
          <Fab
            className="page-button"
            onClick={() => handlePageButton("previous")}
          >
            <i className="material-icons md-48">{PAGE_BUTTON_ICON_PREVIOUS}</i>
          </Fab>
        )}
      </Grid>
      <Grid item>
        {pageDetails && pageIndex !== Object.keys(pageDetails).length - 1 && (
          <Fab className="page-button" onClick={() => handlePageButton("next")}>
            <i className="material-icons md-48">{PAGE_BUTTON_ICON_NEXT}</i>
          </Fab>
        )}
      </Grid>
    </Grid>
  );
}

PageChangeButtonComponent.propTypes = {
  pageDetails: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.string)),
  pageIndex: PropTypes.number,
  handlePageButton: PropTypes.func,
};

const mapStateToProps = (state) => {
  return {
    pageDetails: state.weatherReducer.data.pageDetails,
  };
};

export default connect(mapStateToProps, null)(PageChangeButtonComponent);
