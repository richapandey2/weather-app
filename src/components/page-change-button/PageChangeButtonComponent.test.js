import React from "react";
import { render, cleanup } from "@testing-library/react";
import PageChangeButtonComponent from "./PageChangeButtonComponent";
import { testStore } from "../../utils/test.util";

afterEach(cleanup);

test("should test page change button component", () => {
  const store = testStore();
  const wrapper = render(<PageChangeButtonComponent store={store} />);
  expect(wrapper.asFragment).toMatchSnapshot();
});
