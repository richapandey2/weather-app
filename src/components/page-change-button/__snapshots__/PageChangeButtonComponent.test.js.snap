// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test page change button component 1`] = `[Function]`;
